gateways/checkoutcom/request_builder.go

```diff
@@ -0,0 +1,61 @@
+package checkoutcom
+
+import (
+	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/common"
+	checkout_com_common "github.com/checkout/checkout-sdk-go/common"
+	"github.com/checkout/checkout-sdk-go/payments"
+)
+
+func buildChargeParams(authRequest *sleet.AuthorizationRequest) (*payments.Request, error) {
+	var source = payments.CardSource{
+		Type: "card",
+		Number: authRequest.CreditCard.Number,
+		ExpiryMonth: uint64(authRequest.CreditCard.ExpirationMonth),
+		ExpiryYear: uint64(authRequest.CreditCard.ExpirationYear),
+		Name: authRequest.CreditCard.FirstName + " " + authRequest.CreditCard.LastName,
+		CVV: authRequest.CreditCard.CVV,
+		BillingAddress: &checkout_com_common.Address{
+			AddressLine1: common.SafeStr(authRequest.BillingAddress.StreetAddress1),
+			AddressLine2: common.SafeStr(authRequest.BillingAddress.StreetAddress2),
+			City:         common.SafeStr(authRequest.BillingAddress.Locality),
+			State:        common.SafeStr(authRequest.BillingAddress.RegionCode),
+			ZIP:          common.SafeStr(authRequest.BillingAddress.PostalCode),
+			Country:      common.SafeStr(authRequest.BillingAddress.CountryCode),
+		},
+	}
+
+	return &payments.Request{
+		Source:   source,
+		Amount:   uint64(authRequest.Amount.Amount),
+		Capture:  common.BPtr(false),
+		Currency: authRequest.Amount.Currency,
+		Reference: *authRequest.ClientTransactionReference,
```

Comment:
If this is for order reference, then I think we might prefer to use `MerchantOrderReference` (Cart display ID) than `ClientTransactionReference` here? 
cc @dbook13

Reply:
Agreed. Ideally we would be able to pass both (even if one is in metadata or something), but the merchant will most likely need MerchantOrderReference passed somewhere in order to reconcile orders between their OMS and PSP.

Reply:
Yeah they only have one field. Unless we want to do something funky like merging the strings, I'll replace with MerchantOrderReference